### DISCUSSION TITLE
Pin app window to screen type (e.g. internal retina)

### DISCUSSION THREAD

**Discussion by bric3** [rxhanson/Rectangle#1148] [Ideas]
I have a laptop and I move offices, some have two external screens, some have a single bigger external screen.

The main display

* If there are 2 external screen, the _main display_ is the macbook one.
* If there is a single external screen, the _main screen_ is the external one.

However some apps like Slack that I'd like to keep on the macbook retina screen are actually placed on the main screen which changes depending in which office I am.

I think it would be nice if Rectangle could pin some apps to the screen type, whether the screen is a main display or an external one. E.g. pinning Slack to the _retina display_.

_Of course there are some tricks to handle, e.g. whether the display is actually connected or active (lid closed). Possibility to clear the remembered pins._

**Comment by rxhanson** [comment]
Thanks for the idea. I don't have the bandwidth to add something like this into Rectangle, but would be fine merging a good pull request that didn't add anything new to the UI (configurable only by Terminal command).

**Reply by bric3** [reply]
Oh I didn't knew there was a cli tool!

I understand well the challenges of OSS developers. Thank you for considering the idea!

**Reply by bric3** [reply]
@rxhanson Hello there, I noticed there's now a Rectangle Pro, could it be something to consider for the pro app ?

**Reply by rxhanson** [reply]
Thanks for asking. Per-display settings is an item on my backlog for Rectangle Pro, but it's not a small one and I don't have a timeframe for it yet.
